nav2_dwb_controller/dwb_plugins/src/xy_theta_iterator.cpp

```diff
@@ -48,55 +51,68 @@ void XYThetaIterator::initialize(
   kinematics_handler_ = kinematics;
 
   nav2_util::declare_parameter_if_not_declared(
-    nh,
-    plugin_name + ".vx_samples", rclcpp::ParameterValue(20));
+    nh, plugin_name + ".vx_samples",
+    rclcpp::ParameterValue(20));
   nav2_util::declare_parameter_if_not_declared(
-    nh,
-    plugin_name + ".vy_samples", rclcpp::ParameterValue(5));
+    nh, plugin_name + ".vy_samples",
+    rclcpp::ParameterValue(5));
   nav2_util::declare_parameter_if_not_declared(
-    nh,
-    plugin_name + ".vtheta_samples", rclcpp::ParameterValue(20));
+    nh, plugin_name + ".vtheta_samples", rclcpp::ParameterValue(20));
 
   nh->get_parameter(plugin_name + ".vx_samples", vx_samples_);
   nh->get_parameter(plugin_name + ".vy_samples", vy_samples_);
   nh->get_parameter(plugin_name + ".vtheta_samples", vtheta_samples_);
 }
 
 void XYThetaIterator::startNewIteration(
-  const nav_2d_msgs::msg::Twist2D & current_velocity,
-  double dt)
+  const nav_2d_msgs::msg::Twist2D & current_velocity, double dt)
 {
   KinematicParameters kinematics = kinematics_handler_->getKinematics();
   x_it_ = std::make_shared<OneDVelocityIterator>(
-    current_velocity.x,
-    kinematics.getMinX(), kinematics.getMaxX(),
+    current_velocity.x, kinematics.getMinX(), kinematics.getMaxX(),
     kinematics.getAccX(), kinematics.getDecelX(), dt, vx_samples_);
   y_it_ = std::make_shared<OneDVelocityIterator>(
-    current_velocity.y,
-    kinematics.getMinY(), kinematics.getMaxY(),
+    current_velocity.y, kinematics.getMinY(), kinematics.getMaxY(),
     kinematics.getAccY(), kinematics.getDecelY(), dt, vy_samples_);
   th_it_ = std::make_shared<OneDVelocityIterator>(
-    current_velocity.theta,
-    kinematics.getMinTheta(), kinematics.getMaxTheta(),
-    kinematics.getAccTheta(), kinematics.getDecelTheta(),
-    dt, vtheta_samples_);
+    current_velocity.theta, kinematics.getMinTheta(),
+    kinematics.getMaxTheta(), kinematics.getAccTheta(),
+    kinematics.getDecelTheta(), dt, vtheta_samples_);
   if (!isValidVelocity()) {
     iterateToValidVelocity();
   }
 }
 
-bool XYThetaIterator::isValidVelocity()
+bool XYThetaIterator::isValidSpeed(double x, double y, double theta)
 {
-  return kinematics_handler_->isValidSpeed(
-    x_it_->getVelocity(), y_it_->getVelocity(),
-    th_it_->getVelocity());
+  KinematicParameters kinematics = kinematics_handler_->getKinematics();
+  double vmag_sq = x * x + y * y;
+  if (kinematics.getMaxSpeedXY() >= 0.0 &&
+    vmag_sq > kinematics.getMaxSpeedXY_SQ() + EPSILON)
+  {
+    return false;
+  }
+  if (kinematics.getMinSpeedXY() >= 0.0 &&
+    vmag_sq + EPSILON < kinematics.getMinSpeedXY_SQ() &&
+    kinematics.getMinTheta() >= 0.0 &&
+    fabs(theta) + EPSILON < kinematics.getMinTheta())
+  {
+    return false;
+  }
+  if (vmag_sq == 0.0 && th_it_->getVelocity() == 0.0) {
```

Comment:
Is it intentional that this is back?

Reply:
You never clarified what you wanted to have done here. The impression I was left was that you wanted to make the check to be in the plugins so you could make a new plugin to make your own decisions about what's valid. If you want to change this behavior, you'll have to let us know what you want to do here instead.